File: 백준/Silver/11723. 집합/집합.cs
using System;
using System.IO;

namespace AlgorithmStudy
{
    class boj11723
    {
        public static void Main(string[] args)
        {
            StreamReader sr = new StreamReader(new BufferedStream(Console.OpenStandardInput()));
            StreamWriter sw = new StreamWriter(new BufferedStream(Console.OpenStandardOutput()));
            bool[] sets = new bool[20];
            int m = int.Parse(sr.ReadLine());
            for (; m > 0; m--)
            {
                string[] input = sr.ReadLine().Split();
                switch (input[0])
                {
                    case "add":
                        sets[short.Parse(input[1]) - 1] = true;
                        break;
                    case "remove":
                        sets[short.Parse(input[1]) - 1] = false;
                        break;
                    case "check":
                        if (sets[short.Parse(input[1]) - 1])
                            sw.WriteLine("1");
                        else
                            sw.WriteLine("0");
                        break;
                    case "toggle":
                        sets[short.Parse(input[1]) - 1] = !sets[short.Parse(input[1]) - 1];
                        break;
                    case "all":
                        for (short i = 0; i < 20; i++)
                            sets[i] = true;
                        break;
                    case "empty":
                        for (short i = 0; i < 20; i++)
                            sets[i] = false;
                        break;
                }
            }
            sr.Close();
            sw.Close();
        }
    }
}

File: 백준/Silver/11723. 집합/집합.cc
#include <bits/stdc++.h>
using namespace std;

int main() {
	ios_base::sync_with_stdio(false);
	cin.tie(NULL);

	bool sets[20];
	memset(sets, false, sizeof(bool) * 20);

	int m;
	cin >> m;
	while (true) {
		string cmd;
		cin >> cmd;

		if (cmd == "all") memset(sets, true, sizeof(bool) * 20);
		else if (cmd == "empty") memset(sets, false, sizeof(bool) * 20);
		else {
			int x;
			cin >> x;
			
			if (cmd == "add") sets[x - 1] = true;
			else if (cmd == "remove") sets[x - 1] = false;
			else if (cmd == "check") printf(sets[x - 1] ? "1\n" : "0\n");
			else if (cmd == "toggle") sets[x - 1] = !sets[x - 1];
		}

		m--;
		if (m == 0) break;
	}

	return 0;
}